devices/HumanLogger/HumanLogger.cpp

```diff
@@ -0,0 +1,463 @@
+                                           "human_state::joint_positions");
+            pImpl->bufferManager.push_back(pImpl->jointVelocitiesInterface,
+                                           timeNow,
+                                           "human_state::joint_velocities");
+        }
+    }
+
+    if (pImpl->settings.logHumanDynamics)
+    {
+        if(!pImpl->iHumanDynamics) {
+            yError() << logPrefix << "The IHumanDynamics pointer is null in the driver loop.";
+            askToStop();
+            return;
+        }
+
+        pImpl->jointTorquesInterface = pImpl->iHumanDynamics->getJointTorques();
+
+        if (pImpl->loggerType == LoggerType::MATLAB) {
+            pImpl->bufferManager.push_back(pImpl->jointTorquesInterface,
+                                           timeNow,
+                                           "human_dynamics::joint_torques");
+        }
+    }
+}
+
+// ======================
+// DeviceDriver interface
+// ======================
+
+bool HumanLogger::open(yarp::os::Searchable& config)
+{
+    std::lock_guard<std::mutex> guard(pImpl->loggerMutex);
+    if (!config.check("period")) {
+        yInfo() << logPrefix << "Using default period: " << DefaultPeriod << "s";
+    }
+
+    const double period = config.check("period", yarp::os::Value(DefaultPeriod)).asFloat64();
+    setPeriod(period);
+
+    // Load settings in the class
+    bool ok = pImpl->loadSettingsFromConfig(config);
+    if (!ok) {
+        yError() << logPrefix << "Problem in loading settings from config.";
+        return false;
+    }
+
+    return true;
+}
+
+void HumanLogger::impl::setLoggerType(std::string& str)
+{
+    if (!std::strcmp(str.c_str(), "matlab")) {
+        this->loggerType = LoggerType::MATLAB;
+    }
+}
+
+bool HumanLogger::impl::loadSettingsFromConfig(yarp::os::Searchable& config)
+{
+    // Check for logLevel parameter
+    this->loggerType = LoggerType::NONE;
+    if (!(config.check("LoggerType")
+          && (config.find("LoggerType").isString() || config.find("LoggerType").isList()))) {
+        yInfo() << logPrefix << "Using default LoggerType : MATLAB";
+        this->loggerType = LoggerType::MATLAB;
+    }
+    else if (config.check("LoggerType") && config.find("LoggerType").isList()) {
+        yarp::os::Bottle* loggerTypeList = config.find("LoggerType").asList();
+
+        for (size_t i = 0; i < loggerTypeList->size(); i++) {
+            std::string option = loggerTypeList->get(i).asString();
+
+            setLoggerType(option);
+        }
+    }
+    else if (config.check("LoggerType") && config.find("LoggerType").isString()) {
+        std::string option = config.find("LoggerType").asString();
+        setLoggerType(option);
+    }
+
+    // Display the current logger level
+    switch (this->loggerType) {
+        case LoggerType::MATLAB: {
+            yInfo() << logPrefix << "LoggerType set to MATLAB";
+            break;
+        }
+        default:
+            break;
+    }
```

Comment:
Can you explain the expected functioning of the "LoggerType" parameter?

With the first `if` the logger type is set to matlab even if the user does not put a valid type (e.g. int).
With the `else if` if a list is provided, only the last values is actually used since the loggerType attribute can hold only a single value.
With the last `else if` the logger type is set to the single string passed as parameter, but no checks are made in the [setLoggerType](https://github.com/robotology/human-dynamics-estimation/pull/298/files#diff-3ef6622f02ff3fd6f397d0111ec631fd3f682d0a00634f6ea06559086dea92b1R182-R187) method.

Also the switch does no action in the default case.
So, if the user sets a parameter different from "matlab", the module will start but will not do anything.

I think it's better if the supported type for the parameter is string and checks are made on that value.

Reply:
Changed in https://github.com/robotology/human-dynamics-estimation/pull/298/commits/229ea4d48fb4762bd7cabaa8067c5b50470a5b5a, a simple string is used to define the logger type (that for the time being is only `matlab`)